src/raster_programs.rs

```diff
@@ -0,0 +1,72 @@
+use crate::{errors::*, Dataset};
+use libc::{c_char, c_int};
+use std::ffi::CString;
+use std::{
+    borrow::Borrow,
+    ptr::{null, null_mut},
+};
+
+/// Build a VRT from a list of datasets.
+/// Wraps [GDALBuildVRT].
+/// See the [program docs] for more details.
+///
+/// [GDALBuildVRT]: https://gdal.org/api/gdal_utils.html#gdal__utils_8h_1a057aaea8b0ed0476809a781ffa377ea4
+/// [program docs]: https://gdal.org/programs/gdalbuildvrt.html
+pub fn build_vrt<D: Borrow<Dataset>>(
+    dest: Option<String>,
+    datasets: &[D],
+    args: Vec<String>,
+) -> Result<Dataset> {
+    // Convert dest to raw string
+    let dest = match dest {
+        Some(s) => CString::new(s)?.into_raw(),
+        None => null_mut(),
+    };
+
+    // Convert args to raw strings
+    let mut c_args = Vec::<*mut c_char>::new();
+    for arg in args {
+        c_args.push(CString::new(arg)?.into_raw());
+    }
+    c_args.push(null_mut()); // Null-terminate it
+
```

Comment:
Something like `let c_options_ptrs = std::iter::chain(args.iter().map(|o| o.as_ptr()), std::iter::once(null_mut())).collect::<Vec<_>>(); ` would be nicer here because you don't have to free them afterwards.